### PR TITLE
DDCYLS-2332 | Migrate frontend to play-frontend-hmrc

### PR DESCRIPTION
**Pull request by jpatton638**
Migrate entire frontend to play-frontend-hmrc

### CODE REVIEW

File: app/controllers/businessdetails/ActivityStartDateController.scala [hunk 1]
```diff
@@ -19,65 +19,67 @@ package controllers.businessdetails
...
-              } yield bm
 
-              val businessType = for {
-                bt <- ControllerHelper.getBusinessType(businessMatching)
-              } yield bt
+      formProvider().bindFromRequest().fold(
+        formWithErrors => Future.successful(BadRequest(activity_start_date(formWithErrors, edit))),
+        data => {
+          dataCache.fetchAll(request.credId) flatMap { maybeCache =>
+            val businessMatching = for {
```
[the remark this answers is not in the document]
Comment (jpatton638) [reply app/controllers/businessdetails/ActivityStartDateController.scala:60]: I just did a copy paste job from the old code but this looks a lot better, cheers!

File: app/controllers/msb/ExpectedThroughputController.scala [hunk 2]
```diff
@@ -36,36 +35,38 @@ class ExpectedThroughputController @Inject() (authAction: AuthAction,
...
     implicit request => {
-      Form2[ExpectedThroughput](request.body) match {
-        case f: InvalidForm => Future.successful(BadRequest(expected_throughput(f, edit)))
-        case ValidForm(_, data) =>
+      formProvider().bindFromRequest().fold(
+        formWithErrors => Future.successful(BadRequest(view(formWithErrors, edit))),
+        data =>
           for {
             msb <- dataCacheConnector.fetch[MoneyServiceBusiness](request.credId, MoneyServiceBusiness.key)
             _ <- dataCacheConnector.save[MoneyServiceBusiness](request.credId, MoneyServiceBusiness.key,
```
[the remark this answers is not in the document]
Comment (jpatton638) [reply app/controllers/msb/ExpectedThroughputController.scala:61]: It is retrieving the existing MoneyServiceBusiness instance, copying the case class to change the `throughput` field to the value that the user submitted using the play Form, then saving it

File: app/controllers/msb/SendMoneyToOtherCountryController.scala [hunk 3]
```diff
@@ -18,73 +18,77 @@ package controllers.msb
...
-                    .mostTransactions(None)
+  def post(edit: Boolean = false): Action[AnyContent] = authAction.async {
+    implicit request => {
+      formProvider().bindFromRequest().fold(
+        formWithErrors =>
+          Future.successful(BadRequest(view(formWithErrors, edit))),
+        data =>
+          dataCacheConnector.fetchAll(request.credId) flatMap { maybeCache =>
+            val result = for {
+              cache <- maybeCache
```
[the remark this answers is not in the document]
Comment (jpatton638) [reply app/controllers/msb/SendMoneyToOtherCountryController.scala:62]: They are both pages with a similar design and data types on the same journey, but the page, form provider and data they deal with is different

File: app/controllers/responsiblepeople/ApprovalCheckController.scala [hunk 4]
```diff
@@ -18,49 +18,47 @@ package controllers.responsiblepeople
...
-  implicit val boolWrite = utils.BooleanFormReadWrite.formWrites(FIELD_NAME)
-  implicit val boolRead = utils.BooleanFormReadWrite.formRule(FIELD_NAME, "error.required.rp.approval_check")
-
-  def get(index: Int, edit: Boolean = false, flow: Option[String] = None) = authAction.async {
+  def get(index: Int, edit: Boolean = false, flow: Option[String] = None): Action[AnyContent] = authAction.async {
     implicit request =>
       getData[ResponsiblePerson](request.credId, index) map {
-        case Some(ResponsiblePerson(Some(personName),_,_,_,_,_,_,_,_,_,_,_,_,_,_,alreadyPassed,_,_,_,_,_,_)) if (alreadyPassed.hasAlreadyPaidApprovalCheck.isDefined) =>
-          Ok(approval_check(Form2[Boolean](alreadyPassed.hasAlreadyPaidApprovalCheck.get), edit, index, flow, personName.titleName))
+        case Some(ResponsiblePerson(Some(personName),_,_,_,_,_,_,_,_,_,_,_,_,_,_,ApprovalFlags(_, Some(hasAlreadyPaidApprovalCheck)),_,_,_,_,_,_))  =>
```
[the remark this answers is not in the document]
Comment (jpatton638) [reply app/controllers/responsiblepeople/ApprovalCheckController.scala:44]: Only issue with this is that the form will always pre-populate with false selected, I'll rework it another way to make it cleaner

File: app/controllers/renewal/AMLSTurnoverController.scala [hunk 5]
```diff
@@ -40,34 +37,37 @@ class AMLSTurnoverController @Inject()(val dataCacheConnector: DataCacheConnecto
...
-            } yield Form2[AMLSTurnover](turnover)) getOrElse EmptyForm
-            Ok(amls_turnover(form, edit, businessMatching.alphabeticalBusinessActivitiesLowerCase(false)))
-          }) getOrElse Ok(amls_turnover(EmptyForm, edit, None))
+      dataCacheConnector.fetchAll(request.credId) map { optionalCache =>
+        (for {
+          cache <- optionalCache
+          businessMatchingOpt = cache.getEntry[BusinessMatching](BusinessMatching.key)
+          businessMatching <- businessMatchingOpt
+          form = getForm(businessMatchingOpt)
+          activities = businessMatching.alphabeticalBusinessActivitiesLowerCase(false)
```
[the remark this answers is not in the document]
Comment (jpatton638) [reply app/controllers/renewal/AMLSTurnoverController.scala:51]: Really good shout, I've gone with a `flatMap` over using `head`, eg:

```
      val futureFormFillBusinessActivities: Future[Form[AMLSTurnover]] = renewalService.getFirstBusinessActivityInLowercase(request.credId).map(formProvider(_))
      val futureAMLSTurnover: Future[Option[AMLSTurnover]] = renewalService.getRenewal(request.credId).map(_.turnover)

      for {
        optBm <- renewalService.getBusinessMatching(request.credId)
        form <- futureFormFillBusinessActivities
        optTurnover <- futureAMLSTurnover
      } yield {
        val filledFormOrEmpty = optTurnover.map(turnover => form.fill(turnover)).getOrElse(form)
        Ok(view(filledFormOrEmpty, edit, optBm.flatMap(_.alphabeticalBusinessActivitiesLowerCase())))
      }
```

**Review by lewishay — APPROVED** [review]

**Pull request merged**